rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -39,6 +39,24 @@
 
 namespace YAML
 {
+/// Allow pass a metadata version to the sub-structs of BagMetadata for deserializing.
+/**
+  * Encoding should always use the current metadata version, so it does not need this value.
+  * We cannot extend the YAML::Node class to include this, so we must call it
+  * as a function with the node as an argument.
+  */
+template<typename T>
+T versioned_decode(const Node & node, int version)
```

Comment:
`decode_for_version` ? Not clear what this means.

Reply:
Sure, done